### ISSUE TITLE
LogError does not emit exception details in Azure Function

### ISSUE THREAD

**Issue by lostmsu** [Azure/azure-functions-host#6476]
I am using injected `ILogger log` as follows in my Azure Function:

```csharp
catch (InvalidOperationException ex) {
   log.LogError(ex, "from " + source);
   return new UnauthorizedResult();
}
```

When I open the function in Azure Portal, and switch to "Code + Test", when this line hits all I get in the "Logs" is

```
2020-07-29T16:59:44Z   [Error]   from x.y.z.a
```

E.g. no exception message and stack trace do not actually appear in the log.

#### Repro steps

1. Create an Azure Function 3.0 with C#, that receives `ILogger log`

2. Throw and catch an exception `error`

3. Log the error using `log.LogError(error, "Custom message")`

4. Deploy the function, and open Code + Test in Azure Portal

5. Expand "Logs"

6. Hit the function, provoking the error to be logged

#### Expected behavior

The message printed in "Logs" section of Azure Portal should have exception details (message, stack trace) in addition to the custom message

#### Actual behavior

Only custom message is printed

#### Known workarounds

Manually attach exception details to the custom message.

#### Related information 

Packages:
```xml
<PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.0.0" />
<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="3.0.3" />
<PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.3" />
```

See also: 1 year old [question on StackOverflow](https://stackoverflow.com/questions/56060604/ilogger-logerror-not-logging-exception-details).

P.S. I can not upgrade SDK due to another issue with `CloudTable` binding

**Comment by ghost** [comment]
Looking forward to updates on this

**Comment by davghouse** [comment]
Looking forward to updates on this

**Comment by PingPongSet** [comment]
Do we have a fix for the bug or the roadmap for fixing it? I have a similar issue.

**Comment by Sti2nd** [comment]
Seems to be working now (with Azure Application Insights)! When passing in the exception to ILogger the log is considered an exception instead of a trace. In Azure Application Insights, instead of querying for `traces` you must now query for `exceptions`
![image](https://user-images.githubusercontent.com/6373430/137913770-b588ceca-e531-4e87-a579-fb961ef53a9a.png)


Relevant documentation: https://docs.microsoft.com/en-us/azure/azure-monitor/app/ilogger#what-application-insights-telemetry-type-is-produced-from-ilogger-logs-where-can-i-see-ilogger-logs-in-application-insights

**Comment by v-bbalaiagar** [comment]
Hi @lostmsu , Let us know if the above information resolves your query, or if you have any further queries.

**Comment by mirekkukla** [comment]
@v-bbalaiagar the issue continues to exist as described in the issue description.

In reference to @Sti2nd 's comment: the issue is not that stacktraces don't show up in application insights: they do (and have for a long time, at least since I've started following this ticket).

Rather, the problem is that the stacktraces do not show up in the "logs" section when monitoring a given function (or, I might add, in the "invocation details" panel, as shown below):

![image](https://user-images.githubusercontent.com/903370/150621971-d3af1b67-66c2-4720-8c11-594d7d473949.png)

The red box shows logger output where I logged an exception as follows:
```
logger.LogError(ex, "Testing if exception logging works now")
```
Note that the exception stacktrace is _not_ being displayed.

The green box shows logger output where I explicitly logged the exception contents myself, as this is my current workaround:
```
// Azure doesn't log exception details, log the stacktrace explicitly
// https://github.com/Azure/azure-functions-host/issues/6476
logger.LogError(ex.ToString());
```
